packages/runtime/src/components/builtin/Text/EditableText.tsx

```diff
@@ -171,7 +174,22 @@ const EditableText = forwardRef(function EditableText(
     if (event.relatedTarget == null) editor.select(selection)
   }, [])
 
-  const isInBuilder = useIsInBuilder()
+  // If the user selects any text when they are in interaction mode or content mode,
+  // deselect any text when they go to the build mode.
+  useEffect(() => {
+    if (editMode === EditModes.BUILD) editor?.deselect().blur()
+  }, [editMode, editor])
+
+  // HACK: Slate holds on to the very first DOM event handlers passed in and doesn't update them
+  // even if they change.
+  const lastEditMode = useRef(editMode)
+  if (lastEditMode.current !== editMode) lastEditMode.current = editMode
+
+  const handleClick = (event: ReactMouseEvent, _editor: Editor, next: () => any) => {
+    next()
+
+    if (lastEditMode.current === EditModes.CONTENT) event.stopPropagation()
+  }
```

Comment:
This is going to cause problems with Slate when we attempt to extend our rich text control to support rich inline plugins. Also, this seems to be very specific to the "click" event when it seems this should be handled more generically, like when the contenteditable is focused? Are there other interactions like this one where we want to prevent stuff like this?

I don't have a better solution for this right now so maybe we just keep this, but we should add a comment as to why we have this for our future selves.